prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -341,41 +341,61 @@ class DetailedSubmissionHistory(
                 // Check if destination has already been added
                 // if it is increase item counts
                 // otherwise add it to destinations
-                val index = destinations.indexOf(descendantDest)
-                if (index >= 0) {
-                    destinations[index].itemCount += descendantDest.itemCount
-                    destinations[index].itemCountBeforeQualFilter =
-                        destinations[index].itemCountBeforeQualFilter?.plus(
-                            descendantDest.itemCountBeforeQualFilter ?: 0
-                        )
-                } else {
-                    destinations += descendantDest
+                with(destinations) {
+                    val index = indexOfFirst {
```

Comment:
This is OK, but you could use a filter or firstOrNull instead of dealing with the index number

Reply:
Using firstOrNull now